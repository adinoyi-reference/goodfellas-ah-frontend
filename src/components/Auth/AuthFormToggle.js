import React, { Component } from 'react';
import { Link } from 'react-router-dom';

// eslint-disable-next-line react/prefer-stateless-function
class AuthFormToggle extends Component {
  render() {
    const { children: { props: { formtype } } } = this.props;
    const activeSignin = (formtype === 'signin' ? 'active' : null);
    const activeSignup = (formtype === 'signup' ? 'active' : null);
    return (
      <div className="auth-buttons">
        <Link className="both-links" to="/auth/signup">
          <button type="button" className={`buttons sign-up ${activeSignup}`}>SIGN UP</button>
        </Link>
        <Link className="both-links" to="/auth/signin">
          <button type="button" className={`buttons sign-in ${activeSignin}`}>SIGN IN</button>
        </Link>
      </div>
    );
  }
}

export default AuthFormToggle;
