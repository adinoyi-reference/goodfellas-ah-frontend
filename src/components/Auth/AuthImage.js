import React from 'react';
import author from '../../assets/author.jpg';

const AuthImage = () => (
  <div className="author-writing">
    <p className="auth-header">Authors Haven</p>
    <img width="84%" src={author} alt="" />
    <p className="auth-title">A community of like minded authors</p>
  </div>
);

export default AuthImage;
