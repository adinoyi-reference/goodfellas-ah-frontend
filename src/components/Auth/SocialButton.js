import React from 'react';

const SocialButton = ({ image, className, type }) => (
  <div className="social-link">
    <a href={`http://localhost:3000/api/auth/${type}`}>
      <div className={className}>
        <img src={image} alt="" />
      </div>
    </a>
  </div>
);

export default SocialButton;
