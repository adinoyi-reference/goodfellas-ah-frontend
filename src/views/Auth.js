import React, { Component } from 'react';
import SigninForm from '../containers/SigninForm';
import SignupForm from '../containers/SignupForm';
import facebook from '../assets/facebook-image.png';
import google from '../assets/google-image.png';
import twitter from '../assets/twitter-image.png';

class Auth extends Component {
  state={
    active: 'signin',
  }

  changeForm(formType) {
    this.setState({ active: formType });
  }

  renderAuthForm(active) {
    return (active === 'signin' ? <SigninForm {...this.props} /> : <SignupForm {...this.props} />);
  }

  render() {
    const { active } = this.state;
    const activeSignin = (active === 'signin' ? 'auth-button-active' : null);
    const activeSignup = (active === 'signup' ? 'auth-button-active' : null);

    return (
      <div className="auth">
        <section className="auth-image">
          <div className="auth-title">Authors Haven</div>
          <div className="auth-tagline">A Community of Like Minded Authors</div>
        </section>
        <div className="auth-image-divide" />

        <section className="auth-form-wrapper">
          <div>
            <div className="auth-toggle-wrapper">
              <button
                type="button"
                className={`auth-button-toggle ${activeSignup}`}
                onClick={() => this.changeForm('signup')}
              >
                SIGN UP
              </button>
              <button
                type="button"
                className={`auth-button-toggle ${activeSignin}`}
                onClick={() => this.changeForm('signin')}
              >
                SIGN IN
              </button>
            </div>
            {this.renderAuthForm(active)}
            <div className="auth-form-or">Or</div>
            <div className="auth-icon-wrapper">
              <div className="auth-social-icons">
                <div className="auth-social-button">
                  <img src={facebook} alt="facebook signup" />
                </div>
                <div className="auth-social-button">
                  <img src={google} alt="google signup" />
                </div>
                <div className="auth-social-button">
                  <img src={twitter} alt="twitter signup" />
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    );
  }
}
export default Auth;
