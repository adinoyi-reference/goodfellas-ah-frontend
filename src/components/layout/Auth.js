import React from 'react';
import author from '../../assets/author.jpg';
import facebook from '../../assets/facebook-image.png';
import google from '../../assets/google-image.png';
import twitter from '../../assets/twitter-image.png';
import AuthFormToggle from '../AuthFormToggle';
import '../../styles/styles.scss';


const Auth = (props) => {
  const { children } = props;

  return (
    <div className="auth-flex">
      <div className="author-writing">
        <p className="auth-header">Authors Haven</p>
        <img width="84%" src={author} alt="" />
        <p className="auth-title">A community of like minded authors</p>
      </div>
      <div className="auth-form">

        <AuthFormToggle {...props} />

        {children}

        <div className="or">Or</div>

        <div className="social-icons">

          <div className="social-link" to="#">
            <div className="facebook"><img src={facebook} alt="" />
            </div>
          </div>

          <div className="social-link" to="#">
            <div className="google"><img src={google} alt="" />
            </div>
          </div>

          <div className="social-link" to="#">
            <div className="twitter"><img src={twitter} alt="" />
            </div>
          </div>

        </div>
      </div>
    </div>
  );
};

export default Auth;
