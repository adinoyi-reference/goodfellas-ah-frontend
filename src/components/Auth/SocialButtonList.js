import React from 'react';
import facebook from '../../assets/facebook-image.png';
import google from '../../assets/google-image.png';
import twitter from '../../assets/twitter-image.png';
import SocialButton from './SocialButton';


const SocialButtonList = () => (
  <div className="social-icons">
    <SocialButton
      image={facebook}
      type="facebook"
      className="facebook"
    />
    <SocialButton
      image={google}
      type="google"
      className="google"
    />
    <SocialButton
      image={twitter}
      type="twitter"
      className="twitter"
    />
  </div>
);


export default SocialButtonList;
