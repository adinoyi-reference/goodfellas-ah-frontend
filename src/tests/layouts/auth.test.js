import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
// import { MemoryRouter } from 'react-router-dom';
import Auth from '../../components/layout/Auth';
import Root from '../../root';

let wrapped;

beforeEach(() => {
  wrapped = shallow(

    <Root>
      <Auth />
    </Root>,

  );
});
afterEach(() => wrapped.unmount());


describe('Auth layout', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
