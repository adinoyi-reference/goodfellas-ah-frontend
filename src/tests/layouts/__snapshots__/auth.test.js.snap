// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth layout render features container should render as expected 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Auth />
</Provider>
`;
